### ISSUE TITLE
Ongoing Issues with SPZB0001 / Eurotronics Spirit Zigbee

### ISSUE THREAD

**Issue by lephisto** [EUROTRONIC-Technology/Spirit-ZigBee#1]
To Track it down i quickly drop links to the ongoing issues with various Versions of the Heating Thermostat:

https://github.com/Koenkk/zigbee2mqtt/issues/4571

https://github.com/Koenkk/zigbee-herdsman-converters/issues/1893

**Comment by malfroid** [comment]
Other links:

https://github.com/dresden-elektronik/deconz-rest-plugin/issues?q=is%3Aissue+is%3Aopen+eurotronic+spirit

https://github.com/zigpy/zha-device-handlers/issues/689

**Comment by malfroid** [comment]
@Witriol do you have any update or estimated timelines on an improved firmware (beta), or on improving the integration with Z2M, ZHA or deCONZ?

**Comment by Witriol** [comment]
@malfroid The main focus is currently on deCONZ related issues (that is what has Eurotronic approached us with) and ZHA and Z2M will be next in line. First release of improved firmware should be available in one or two weeks together with OTA update wiki guide. Currently I'm collecting information about other issues since it seems that there are more and more issues being reported.

**Comment by hargathor** [comment]
Hello @Witriol 
You can also follow an ongoing issue related to this version here: https://github.com/KiwiHC16/Abeille/issues/1582

In my case this is related to the support of a PiZigate board attached to a raspberry pi but it seems heavily related to the other issue listed above.

**Comment by klangfarbe** [comment]
I am having also the issue, that at least one of my six thermostats constantly breaks by not reacting to anything anymore. The local temperature is way higher (23.5 degree) than the _occupied_heating_setpoint_ (20 degree) and it is still heating. The _pi_heating_demand_ does not react and even changing the temperature directly at the thermostat does not do anything. 

After removing batteries and readapting is is working now but shows a _pi_heating_demand_ of 100 while the real valve is closed which is correct since the local temperature is way higher. Only after a new _occupied_heating_setpoint_ is send to the thermostat, the _pi_heating_demand_ is set correctly to 0. But it's only a matter of hours that it breaks again.

This thermostats firmware is way too broken and urgently needs to be fixed...

**Comment by BeistleA** [comment]
I have nine of these Thermostats running on Openhab.
Coordinator CC2531 20190608
Software Zigbee2Mqtt 1.16.2 commit: 0514204
One Thermostat: 20190408/18181120
Eight Thermostats: 20191014/22190930
I use Osram Plugs as Repeater. The one Thermostat with FW18181120 on it is only running while directly connected to the Coordinator but also not always accepting commands (Timeout error)
The others are all around the house and working nearly perfect.
@Witriol: If I can support with any kind of logs or try something let me know. If more information is needed just ask.

Greetings Andy

**Comment by saxn-paule** [comment]
I have exactly the same issue like @klangfarbe
2 of my 5 thermostats drop once a day. 

I also have an Osram plug next to each thermostat but it doesn't help.

**Comment by pY4x3g** [comment]
Same issue here (CC2531, z2m, home assistant), 2 new thermostats only work for some hours and then completely loose the connection and can only be controlled manually. I stopped counting and reinserting the battery after 15+ times, currently I only use them as "dump" thermostats and control them manually.

**Comment by ChristianHersevoort** [comment]
I have a similar issue where they stop responding for some time, but after a few days they start responding again. Restarting the zigbee2mqtt daemon doesn't seem to have an impact.

Another issue I'm having where 1 of 3 TRV sends event messages (visible in zigbee2mqtt everso often

**Comment by fribse** [comment]
I have 4 TRV's around the house, they are a bit older, connected to deConz, and I use IKEA bulbs / plugs. They accept routing for both xiaomi sensors and the Spirit zigbee.
The biggest problem I have is in the adding process, it is daunting to add these to the deconz, requires forcing read of attributes, and doing things in the exact right order and timing.

But day to day, they work.  Datecode is: 20191014
With the later reported problems, I've not had the nerve to buy more of these.

**Comment by saxn-paule** [comment]
All my TRVs have date code 20191014 and two of them have the connection loss issue.

**Comment by vangguppie** [comment]
I have 6 TRV's all date code 20191014 and all of them have the connection lost issue. 2 of them are still in the box and i am considering returning them because of this issue.

I use HA with deconz (with longer usb cable). Join of TRV's in the same room as deconz went well. When discovered in deconz goto clusterinfo and press read.

latest version of deconz firmware
7 routers (Ledstrips and light bulbs) and aprox. 10 endpoints.

**Comment by klangfarbe** [comment]
**Just a quick update:** Regarding a mail I received from Eurotronic Support, the OTA update should be released within the next 2-3 weeks.

**Comment by byterazor** [comment]
Lol,they are telling this since before christmas. I returned all 7 spirits back to amazon. They are unsuable.

**Comment by heggink** [comment]
That's not entirely correct, we have moved from 1-2 weeks to 2-3 weeks now so the bigger the slippage, the bigger the future time frame :-). Ah well, it will get resolved and my workaround (using the other setpoint) works for now.

**Comment by byterazor** [comment]
I have more problems:

- two do not join zigbee networks at all (tested with two different gateways), and yes i resetted them the hardware multiple times
- all newer versions  have wrong mappings of zigbee attributes (i could cope with that)
- two spirits do not close the valve correctly, so it is hot in my sleeping room even the thermostats target temperature is 16

The first and last are blockers for me, because the PAF (Partner Acceptance Factor) is below zero.

I am doing software development for a living too and this looks like a serious quality management problem.

**Comment by saxn-paule** [comment]
> * two spirits do not close the valve correctly, so it is hot in my sleeping room even the thermostats target temperature is 16

Exactly this behavior two of my TRV also have. Valve position is at 2% but the radiator is hot.

The TRV that wasn't able to join my network is probably bricked now, as long as somebody couldn't tell me how I could leave the OTA mode.
I resetted it multiple times an tried the reset like the Z-Wave Spirit TRV
Hold Boost button, remove battery while stil holding boost button, insert battery and still hold boost button. After a while boost button starts blinking red and never stopped blinking. Also after removing/inserting batteries again it keeps flashing while the display shows nothing.

**Comment by poiuztr123** [comment]
I have 6 TRV's. All with date code 20191014. I was able to pair all of them, but only when I was not too far away from the conbee stick in my raspi. I have 6 innr plugs in the house that act as routers/repeaters and help with very many Xiaomi Aqara devices that are spread across the different levels of our house and they work nicely. However, it seems that pairing and also operating the Eurotronic Spirits via the innr plugs as routers is not possible, at least i do not see a line between them in deconz-gui.
Independently from where they are in the house I loose for 5 of the devices connection quite quickly after repairing. Thermostats are basically unusable in this state. If the firmware update doesn't come quickly I will have to buy different ones.

**Comment by Sjack-Sch** [comment]
I have 2 Spirits zigbee. One has the issue that is stops communicating after a few hours. It needs to get a factory reset before it starts communicating again and then stops after a few hours. It is now used as a dumb manual operated device.

For the second one the communication works OK, however the battery value does not work, always reported as zero, while the battery is a new full alkaline battery.
It also reports `"eurotronic_error_status":1˜` 
Does anybody know what this error status means, is this maybe related to the battery value zero issue?

`{
  "battery":0,
  "boost":false,
  "child_protection":false,
  "current_heating_setpoint":9.5,
  "eurotronic_error_status":1,
  "eurotronic_host_flags":{
    "boost":false,
    "child_protection":false,
    "mirror_display":true,
    "window_open":false
  },
  "eurotronic_system_mode":3,
  "linkquality":68,
  "local_temperature":12.85,
  "mirror_display":false,
  "occupied_heating_setpoint":9.5,
  "pi_heating_demand":100,
  "system_mode":"auto",
  "unoccupied_heating_setpoint":10,
  "window_open":false
}`

**Comment by Howaner** [comment]
I can't understand why the vendor / service provider doesn't use the issue functionality in github to track down the issues like an ynormal project does ..
Then each issue could be created as a seperate issue and the service provider can close it when it's fixed.
This way we could see the status and it would be much easier to track down the issues.

**Comment by gitolicious** [comment]
Isn't that something you=the users would need to, or at least could do? I was also overwhelmed by this long, unstructured list of "X is not working for me", "Y is not working for me". This definitely needs to be broken down into separate issues.

**Comment by heggink** [comment]
Not really. If you buy a commercial product, it should work out be fixed under support. This is neither open source not Kickstarter.

**Comment by heggink** [comment]
Does this firmware bring back the original (reasonably working) functionality then?

**Comment by Howaner** [comment]
> I have added an already released firmware with date code [20190408](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases/tag/20190408) that I have received from Eurotronic for public release. It predates the problematic "double" release with same datecode.
> 
> Another update is that in February Eurotronic will have an employee for maintaining this GitHub page.

Thank you for the guide and the firmware file.
I tried to do OTA updates on my thermostat (Broken version without the colorful packaging) but it looks like the thermostat ignores the OTA requests.
After a click on "Update", the progress goes to "Queued" and remains at "Queued", even after hours. When I reset the device by pulling out the batteries, the progress goes back to "Idle"

**Comment by hackex** [comment]
> I have added an already released firmware with date code [20190408](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases/tag/20190408) that I have received from Eurotronic for public release. It predates the problematic "double" release with same datecode.
> 
> Another update is that in February Eurotronic will have an employee for maintaining this GitHub page.

Do you know if it is possible to flash the firmware over ZHA integration of Home Assistant? If not, is there a way to just install deconz and use the conbee 2 stick as it used by Home Assistant and just do the OTA with deconz without any interference and then switch back?

Thank you

**Comment by CreatsheR** [comment]
> After a click on "Update", the progress goes to "Queued" and remains at "Queued", even after hours. When I reset the device by pulling out the batteries, the progress goes back to "Idle"

Similar behavior here. I click on Update and the status changes to Idle. After that nothing happens in DeConz or at/on the thermostat...

**Comment by Witriol** [comment]
@heggink From what I have been told, it should. I have tested problematic attribute 0x4003 and it works fine.

@Howaner That as happened to me once as well, but it got fixed by first clicking Queue and then Update button. Could you check OTAU cluster attributes? It should show current OTA version string (the 20190408 update has OTA string 0x00122C380).

@hackex I have no experience with ZHA integration of Home Assistant, unfortunately. Will ask Eurotronic guys.

**Comment by Sjack-Sch** [comment]
> > I have added an already released firmware with date code [20190408](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases/tag/20190408) that I have received from Eurotronic for public release. It predates the problematic "double" release with same datecode.
> > Another update is that in February Eurotronic will have an employee for maintaining this GitHub page.
> 
> Do you know if it is possible to flash the firmware over ZHA integration of Home Assistant? If not, is there a way to just install deconz and use the conbee 2 stick as it used by Home Assistant and just do the OTA with deconz without any interference and then switch back?
> 
> Thank you

I requested KoenK to add the firmware file to zigbee2mqtt yesterday (2012-01-26). It should be available in the dev version of HA zigbee2mqtt: 
https://github.com/Koenkk/zigbee2mqtt/discussions/2921#discussioncomment-312386

**Comment by CreatsheR** [comment]
> @Howaner That as happened to me once as well, but it got fixed by first clicking Queue and then Update button. Could you check OTAU cluster attributes? It should show current OTA version string (the 20190408 update has OTA string 0x00122C380).

Tried that as well. Doesn't work... The Abort-button doesn't work as well. The Progress Idle stays there "forever".

![Bildschirmfoto 2021-01-28 um 08 52 01](https://user-images.githubusercontent.com/72187841/106106642-23227d80-6146-11eb-85d4-cd6d840d32a4.png)

**Comment by sushifrick** [comment]
@CreatsheR this looks a bit strange to me. My Spirits all have the Vendor ID 0x1037 and Image ID 0x110c. I Updated all of them yesterday. Some were on Version 0.22 before but after the update and the reboot of the devices, which takes a couple of minutes I hat do use the query button to show the "new" old Version number 0.18 (mousover).

Until now the devices do behave, I am looking forward to a new update nevertheless.

**Comment by sushifrick** [comment]
Just to make it more clear. The reboot of the device took a couple of minutes, the ota took a bit over 2 hours per device on default settings (25ms) and about 80-90 minutes with a packet spacing of 15ms.

**Comment by vrabac** [comment]
@Witriol 
> 
> 
> I have added an already released firmware with date code [20190408](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases/tag/20190408) that I have received from Eurotronic for public release. It predates the problematic "double" release with same datecode.
> 

Can you please check why this can be applied with zigbee2mqtt, @Koenkk wrote [some info here](https://github.com/Koenkk/zigbee2mqtt/discussions/2921#discussioncomment-315790) about why this can be applied to my devices.

> Your SPZB0001 requests a different image type than the one on https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases/tag/20190408 (`65535` vs `4364` https://github.com/Koenkk/zigbee-OTA/blob/51a087b0b6baa153ebe55d6d870fee32e6fe8af9/index.json#L699).
> 
> According to the Zigbee spec this means this image is not compatible with this device.

@MrMartin92 [wrote here ](https://github.com/Koenkk/zigbee2mqtt/issues/4571#issuecomment-768905353)also some stuff about same thing:
> 
> 
> Same Problem here with my docker container and `latest-dev`.
> 
> I noticed 65535 is 0xFFFF. For me it doesn't look right. So I checked [zigbee-OTA's index.json](https://github.com/Koenkk/zigbee-OTA/blob/51a087b0b6baa153ebe55d6d870fee32e6fe8af9/index.json#L699) and indeed the imageType here is 4364. So something is wrong. But I have no time to dig deeper at the moment.


My device:
```
"model":"SPZB0001"
"vendor":"Eurotronic"}
"genIdentify","hvacThermostat"
"power_source":"Battery"
"software_build_id":"22190930"

```

PS: I have only zigbee2mqtt (and mqtt (mosquitto)), building an home-assistant in progress).

**Comment by CreatsheR** [comment]
@vrabac I think you meant "can't" instead of "can"!?

Nevertheless, I see some parallels to my Problem. In deCONZ the 5 Spirits that doesn't work properly all show only maximum values after clicking Query. The one Spirit which works properly shows proper values in the Vendor, Image and Version Fields:
![Bildschirmfoto 2021-01-28 um 12 35 15](https://user-images.githubusercontent.com/72187841/106133411-8079f700-6165-11eb-8609-1ca3bb2383f8.png)

**Comment by poiuztr123** [comment]
I tried to update with the firmware posted here with no success (they remain in "idle"). Five of my Spirits show 0xffff. I am using deconz and a conbee, if that helps. Am I doing something wrong?
![image](https://user-images.githubusercontent.com/49477245/106192128-0f116700-61ac-11eb-94f0-7ec93c572650.png)

**Comment by ebaauw** [comment]
The rows in the table are populated from the data the device sends to deCONZ.  I've never seen these 0xFFFF values; the table initialises at 0x0000 and stays on those values in case the device sends no data.  Can you try and read the (client, grey) _OTAU_ cluster attributes?  Particularly interested if the _Manufacturer ID_ and _Current File Version_ show all Fs there as well.  
Would also be interesting to double-check using a Zigbee sniffer whether the device is actually sending the Fs or whether there's something wrong in the OTAU plugin.

If it's really the device sending these values, it might reject the firmware, as the _Manufacturer ID_ and _Image Type_ don't match, or because it thinks it has a newer firmware version than the file.  I've seen devices that allow firmware downgrades (IKEA), but also that don't (Hue).

If you download the firmware file to `~/otau` and restart deCONZ, the OTAU plugin should copy the file to `1037_110C_0122C380.zigbee`.  When you next select the device and press _Query_, the OTAU plugin should match the file to the device.  For end-devices, you need to start _Update_ manually, as you might need to install fresh batteries first (especially for CR2032 button cells).

> the ota took a bit over 2 hours per device on default settings (25ms) and about 80-90 minutes with a packet spacing of 15ms.

That's not bad.  In my experience, routers typically take 10-30 minutes to upgrade the firmware, end-devices might take up to 4 hours.  Note that the device is in control of the upgrade process: it decides when to request the next block.  It helps when the device is in direct range of the OTAU server (i.e. the coordinator).  I never tries messing with the packet spacing.

**Comment by poiuztr123** [comment]
> Can you try and read the (client, grey) OTAU cluster attributes? Particularly interested if the Manufacturer ID and Current File Version show all Fs there as well.

here you go:
![image](https://user-images.githubusercontent.com/49477245/106258518-36a11780-621e-11eb-9c1b-6774caf81d28.png)
![image](https://user-images.githubusercontent.com/49477245/106258634-605a3e80-621e-11eb-99f3-c40d90a860b7.png)

![image](https://user-images.githubusercontent.com/49477245/106258669-6ea85a80-621e-11eb-9ffe-5cabc9f71562.png)

**Comment by CreatsheR** [comment]
> > Can you try and read the (client, grey) OTAU cluster attributes? Particularly interested if the Manufacturer ID and Current File Version show all Fs there as well.
> 
> here you go:

in my deCONZ it looks exactly the same

**Comment by ebaauw** [comment]
Hm, the _Upgrade server_ shows all Fs as well.  That should be the mac address of the coordinator, as returned by the OTAU plugin.  At least it shows the correct _Manufacturer ID_.

The communication between the device and the OTAU plugin seems broken alright.  Without sniffer logs it will be hard to determine what's causing this and whether the OTAU plugin can fix or workaround this.

I'm afraid I cannot be of much further help; my eight TRVs are (were) all on 15181120 (the 20181205 version).  I updated one to 18181120 (20190408) in just under an hour.  ~~I'll be running that firmware for a couple of days before updating the others.~~

I'll be happy to have a deeper look, if I can borrow a spare TRV with the troublesome firmware.  If you can lend me one, please PM me on Discord.

**Edit** Updated all my other TRVs without any issues, each update taking between 50 and 70 minutes.

**Comment by vangguppie** [comment]
> > > Can you try and read the (client, grey) OTAU cluster attributes? Particularly interested if the Manufacturer ID and Current File Version show all Fs there as well.
> > 
> > 
> > here you go:
> 
> in my deCONZ it looks exactly the same

The same here....

**Comment by double1968** [comment]
Could not update too. Same problems

**Comment by dako76** [comment]
Hi, i can not get the firmware file to my raspberry pi to select it in deconz? I installed the homeassistant image on my pi and "ha docker cp" is not working... i copy files via ssh on /root/otau and restart deconz but nothing was copied can anyone give me some help? Otherwise i have to send back Spirit Zigbee :-(

**Comment by vangguppie** [comment]
> Hi, i can not get the firmware file to my raspberry pi to select it in deconz? I installed the homeassistant image on my pi and "ha docker cp" is not working... i copy files via ssh on /root/otau and restart deconz but nothing was copied can anyone give me some help? Otherwise i have to send back Spirit Zigbee :-(

Followed this thread ?

https://community.home-assistant.io/t/update-zigbee-firmware-using-deconz-on-hass-io/157447

**Comment by double1968** [comment]
> Hi, i can not get the firmware file to my raspberry pi to select it in deconz? I installed the homeassistant image on my pi and "ha docker cp" is not working... i copy files via ssh on /root/otau and restart deconz but nothing was copied can anyone give me some help? Otherwise i have to send back Spirit Zigbee :-(

I used this procedure:
with winscp (ssh) I copied the fw in the share folder (located in the root) of Hassio.
Then ,
run this command to copy the OTAU file to the Deconz addon
docker cp /share/20190408_EUROTRONIC_Spirit_Zigbee_0x00122C380.ota addon_core_deconz:/data/otau/

Where 20190408_EUROTRONIC_Spirit_Zigbee_0x00122C380.ota is an example of an OTAU filename.

**Comment by jeankone** [comment]
> > > Can you try and read the (client, grey) OTAU cluster attributes? Particularly interested if the Manufacturer ID and Current File Version show all Fs there as well.
> > 
> > 
> > here you go:
> 
> in my deCONZ it looks exactly the same

exactly the same for me as well

**Comment by fribse** [comment]
> 
> 
> Hi, i can not get the firmware file to my raspberry pi to select it in deconz? I installed the homeassistant image on my pi and "ha docker cp" is not working... 

You have to use the community SSH addon, not the official one.

**Comment by Marlor** [comment]
@ebaauw 
Wenn did you buy yours?
I have 1 from end 2019, working as expected.
And 5 from beginning 2021, all of them reporting fs.
![Deconz](https://user-images.githubusercontent.com/3026213/106382907-eca56680-63c2-11eb-9f7d-43d672a27c5f.PNG)
The nut is different too:
Old:
![old](https://user-images.githubusercontent.com/3026213/106382959-35f5b600-63c3-11eb-9f21-5df498ad085a.jpg)
New:
![new](https://user-images.githubusercontent.com/3026213/106382964-3aba6a00-63c3-11eb-9010-42b626f26d20.jpg)

**Comment by jeankone** [comment]
I can also confirm this.
Mine are from December 2020, reporting all Fs and have also the new nut.

**Comment by ebaauw** [comment]
> Wenn did you buy yours?

First one on 12 Jan 2019 at amazon.de; the next seven on 9 Feb 2019 at yakodo.de.  As I said, they all came with the 20181205 firmware.

**Comment by hasdf** [comment]
> I can also confirm this.
> Mine are from December 2020, reporting all Fs and have also the new nut.

Same here.

**Comment by jeankone** [comment]
after days of trying and trying:
no chance for me to update my 2 thermostats running on 22190930. not getting away from "idle". 
did anybody ever disassembled the devices and checked for a serial port for flashing?

**Comment by solvisit** [comment]
> I'll be happy to have a deeper look, if I can borrow a spare TRV with the troublesome firmware. If you can lend me one, please PM me on Discord.

Did someone send you the xffff device? I have different of them and send to you, I just connect to deconz discord server, but can't find you.

**Comment by solvisit** [comment]
@ebaauw the previous message was for you

**Comment by ebaauw** [comment]
I'm ebaauw#6903 on Discord.

**Comment by ebaauw** [comment]
I received the TRV, thanks.

Sniffing confirms that it sends a _Query Next Image Request_ with _Manufacturer Code_ 0xFFFF, _Image Type_ 0xFFFF, and _File Version_ 0xFFFFFFFF (as shown in the table entry in the OTAU plugin).  After selecting the firmware file and pressing _Update_, the OTAU plugin sends a _Query Next Image Response_ with the proper _Manufacturer Code_ 0x1037, _Image Type_ 0x110C, and _File Version_ 0x0122C380 (from the file). Then the TRV falls silent.

I patched the OTAU plugin to send the response with _Manufacturer Code_ and _Image Type_ 0xFFFF instead, and behold: the TRV starts sending _Image Block Request_ messages for 48 bytes.  The plugin responds with _Image Block Response_ messages with 28 bytes.  However, after three requests/responses the TRV sends an _Upgrade End Request_ and reboots (at least I see a _Rejoin Request_ followed by a Device Announcement_).  Could also be the OTAU plugin requesting a reboot, not sure I'm capturing all packets.  I suspect the TRV checks the file header against the response and rejects the file because the _Manufacturer Code_ and _Image Type_ don't match.

My next attempt will be patching the firmware file instead of the OTAU plugin.

**Comment by jeankone** [comment]
> My next attempt will be patching the firmware file instead of the OTAU plugin.

Your effort and progress on this is awesome! If it would be possible to exchange the file header to somewhat the TRV expects, it should be alright.

**Comment by solvisit** [comment]
thank you ebaauw for your work.

I'm trying to flash the chips too, and maybe you can give me an advice. 

The device is composed by 2 different chips, the zigbee JN5179 and the microcontroller STM8L052c6t6. They communicate by serial port, can be easily divided  and programmed individually. After few test I was able to program with the zigbee chip (with garbage), but not with the microcontroller ( without further study ).

Before going on I'd like to know, from your experience, if it is possible to extract the firmware to flash on the chip from the ota file, and on which chip it has to be written.

I don't care to test and break it.

Thank you

**Comment by ebaauw** [comment]
Without proper documentation, I can only do some (hopefully intelligent) guessing.

As the [JN5179](https://www.nxp.com/products/wireless/zigbee/zigbee-and-ieee802-15-4-wireless-microcontroller-with-512-kb-flash-32-kb-ram:JN5179) already contains a microcontroller, I would guess that the Zigbee application runs on that.  Consequently, the OTA file would contain the firmware for that microcontroller.  _Manufacturer Code_ 0x1037 is for Jennic, which has been acquired by NXP, so that adds up.
 
I would theorise the [STM8L052c6t6](https://www.infinite-electronic.es/datasheet/05-STM8L052C6T6.pdf) controls the thermostat hardware, running the thermostat's PID algorithm.  I guess it makes sense to use a separate ultra-low-power MCU for that, as it probably runs continuously, and you don't want to power up the Zigbee radio all the time.  I wouldn't be surprised if the same component would also be present in the Zwave version of the Spirit.

If you find a way to flash the JN5179 on the Spirit, you would need to extract the firmware from the OTA file.  It contains a 56-byte header, followed by single segment with the firmware.  The segment contains a 2-byte tag, followed by a 4-byte length, so the firmware itself starts at offset 62.  I understand it's encrypted, and it's only decrypted by the device, presumably before flashing.  I wouldn't know if the serial interface would expect to receive the firmware encrypted or unencrypted.  
When trying to get the Hue firmware to work, I (mistakingly) created some corrupted firmware, which was downloaded by the light in full, but it didn't activate it.  I don't know if other Zigbee devices have similar safeguards.

**Comment by ebaauw** [comment]
Unsurprisingly patching the file header runs into the same issue: the Spirit aborts the download after the third block.  Effectively I had patched the file in the OTAU plugin.

Also tried to leave the file header intact, and only patch the _Manufacturer Code_ and _Image Type_ in the Zigbee commands.

Inceased the max block size, so the OTAU plugin now sends 48 bytes (the max size from the Spirit's request).  Now the Spirit sends the abort after the seconds request.  Looking at the timings, it hasn't yet processes the second block.  It seems to choke on something in between offset 28 and 48 in the OTAU header, maybe the description string `DR1175r1v2--JN5179--ENCRYPTED00`?  I tried patching the file putting all `\0`s in the description string, but no joy.

I'm afraid I've run out of ideas what else to try.  In summary, the OTAU function of the Spirit with the 20191014 firmware v22190930 seems broken.  It requests the wrong _Manufacturer Code_ and _Image Type_.  Ik looks like I can workaround that in the OTAU plugin, but then the Spirit aborts the download, presumably because it doesn't like the header.

@Witriol do you have any insights what's happening here?  Can you find out what checks the faulty firmware is doing on the firmware file header?

**Comment by jeankone** [comment]
so in the worst case, we are left with a faulty TRV - which can't even be patched, because it is so faulty, that it wont accept other firmwares....

@solvisit was able to flash the JN5179 with garbage. Can you find out if it accepts the encrypted firmware this way?

**Comment by solvisit** [comment]
Thank you ebaauw for the tips.
@jeankon Flashing the encrypted firmware doesn't work.

**Comment by vrabac** [comment]
@Witriol any news from you in regards of firmware? Even old firmware cannot be applied to a lot of devices..

I don't see much action here on github, you wrote "Another update is that in February Eurotronic will have an employee for maintaining this GitHub page." Is new employee active here?

**Comment by Witriol** [comment]
@vrabac I do not have any update from Eurotronic side since they could not attend last 3 weekly calls and I did not receive any update to my questions via email... So I don't know about the new employee or any future plans, unfortunately. Hopefully tommorrow's call will shed some light and I can write here some new information.

EDIT: The update is that new employee for maintaining GitHub page should start at beggining of March.

**Comment by Witriol** [comment]
> Unsurprisingly patching the file header runs into the same issue: the Spirit aborts the download after the third block. Effectively I had patched the file in the OTAU plugin.
> 
> Also tried to leave the file header intact, and only patch the _Manufacturer Code_ and _Image Type_ in the Zigbee commands.
> 
> Inceased the max block size, so the OTAU plugin now sends 48 bytes (the max size from the Spirit's request). Now the Spirit sends the abort after the seconds request. Looking at the timings, it hasn't yet processes the second block. It seems to choke on something in between offset 28 and 48 in the OTAU header, maybe the description string `DR1175r1v2--JN5179--ENCRYPTED00`? I tried patching the file putting all `\0`s in the description string, but no joy.
> 
> I'm afraid I've run out of ideas what else to try. In summary, the OTAU function of the Spirit with the 20191014 firmware v22190930 seems broken. It requests the wrong _Manufacturer Code_ and _Image Type_. Ik looks like I can workaround that in the OTAU plugin, but then the Spirit aborts the download, presumably because it doesn't like the header.
> 
> @Witriol do you have any insights what's happening here? Can you find out what checks the faulty firmware is doing on the firmware file header?

@ebaauw I have tried patching the firmware binary, but it only lead to OTA update either not even starting or bricking one device (it stayed in bootloader forever). I have access to source code only for Zigbee part (JN5179) and when I have flashed firmware of bricked TRV, it did not recover, so I think that bootloader is placed in ST chip... I will try to do some more digging in the code for the firmware check you have mentioned.

**Comment by univok** [comment]
I run six Eurotronics Spirit Zigbee with date code 20191014 on a Raspberry Pi in Conbee II/deCONZ/homebridge-hue with ioBroker environment. Haven't tried any OTAU update because of this thread. 

Usually, two random of the six become unresponsive during the day, they don't follow any heatpoint change or give the temperature back. The basic and temperatur cluster info cannot be read anymore, and the green/blue dots in deCONZ start flashing in red color from time to time, though the device still shows the green wiring and green flashing dots. ioBroker still shows  the reachable state as True. I don't know what this attribute means, but they aren't reachable at all. Sometimes the wiring in deCONZ is lost too. 

But the devices do not remain in a state where I have to reset them redoing pairing or something similar. Instead, I noticed that I can get them back by pressing "Leave" and "Join" network in the  GUI window. Afters this, everything is fine again. Maybe this can be a workaround for other users too? 

EDIT: This workaround does not help in all cases.  If the wiring disappears in deCONZ, even a reboot of the pi has no effect. A solution can be a reset of the Spirit by pressing the three buttons simultaneously, which set the device in join mode.  Without deleting the sensor, start an "add new sensor" dialogue in Phoscon Web-UI. The Sprit is found again, and the boost button LED flashes in green color. Phoscon fortunately uses the same devices ID after rejoin, but the device name is set to generic "Thermostat" again, which has to be edited afterwards.

**Comment by vrabac** [comment]
@Witriol 
thanks for update regarding GitHub page maintaining.

How about firmware update for other device? See this [comment ](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/issues/1#issuecomment-768987989)above?
Is there any other Firmware from Eurotronic? I also ordered two new devices, and one can't even be paired, other it does pair but have Firmware build date: 20191014 ; Firmware version: 22190930. zigbee2mqtt point this message Failed to call 'OTAUpdate' 'onZigbeeEvent' (AssertionError [ERR_ASSERTION]: No image available for imageType '65535'.

**Comment by iWelna** [comment]
So what we should do now with this new "bad" firmwares which doesn't work correctly? 
Termostat is useless atm, any info from manufacturer @Eurotronics  about replacements or sth like that?

**Comment by mm28ajos** [comment]
@iWelna exectly... @Eurotronics: Any reaction??

**Comment by dako76** [comment]
Hi, i had no success with OTAU i resend this to "Eurotronics" get a new one with firmware "20190408" Date Code. And this one is working perfect! I had no issues over two weeks, no connection failure. I can switch it off, it works as it should. Now i would buy some more :-) 
p.s.: if you get one from Eurotronics and connect it, the led will light red the first time after you finish the install process. You have to wait 5-10 Minutes... than it is working!!

**Comment by magicdude4eva** [comment]
Strangely enough I have Builddate=20191014 / Firmware=22190930 and do not experience any connection issues. I am running ConbeeII with Zigbee2mqtt and have the ConbeeII in the basement and 3 Ikea repeaters (one on ground floor and two on first floor) and since then connectivity is good. 

Flashing Initiated but I aborted since the provided version is older than what I have.

**Comment by iWelna** [comment]
> Hi, i had no success with OTAU i resend this to "Eurotronics" get a new one with firmware "20190408" Date Code. And this one is working perfect! I had no issues over two weeks, no connection failure. I can switch it off, it works as it should. Now i would buy some more :-)
> p.s.: if you get one from Eurotronics and connect it, the led will light red the first time after you finish the install process. You have to wait 5-10 Minutes... than it is working!!


@dako76 How did You process replacement? through warranty with seller or directly with Eurotronic?
I've got ofc common problem cant controll it fully, switch off etc.


@magicdude4eva ts's not about connection issues, they just don't work properly with Hassio, You cant turn them off, only seting temp works and power on. I want to setup some automations but can't coz of that. Before i had z-wave version was perfect but drained a lot of battery.

**Comment by magicdude4eva** [comment]
> @magicdude4eva ts's not about connection issues, they just don't work properly with Hassio, You cant turn them off, only seting temp works and power on. I want to setup some automations but can't coz of that. Before i had z-wave version was perfect but drained a lot of battery.

I am ussing Hassio with AppDaemon and Schedy - with this I automate the schedule - in the example below settings for our bedroom:
![image](https://user-images.githubusercontent.com/1632781/109342288-e85a5500-786b-11eb-898d-044b06a4f938.png)

This is controlled via Schedy/AppDaemon and you see that it matches the schedule:
![image](https://user-images.githubusercontent.com/1632781/109342369-0de75e80-786c-11eb-82c8-b15968bf6c59.png)

**Comment by saxn-paule** [comment]
I also use schedy and in HA it seems to work, but if you have a look at the TVR, sometimes the values don't match.
Window opened, schedy and HA say 6° and the TVR is still on 21°.

Today two of five had exactly this behavior.

**Comment by iWelna** [comment]
With my and the others we cant control them.
I can turn it on, change temp and thats all. Cant change mode or turn off.
I had zwave versions b4 and the were working properly i just sold them some
time ago.

Even when sending mqtt cant turn it off or change mode.

And for automatons im using node red.

pt., 26 lut 2021, 19:53 użytkownik Gerd Naschenweng <
notifications@github.com> napisał:

**Comment by pumuugel82** [comment]
This Spirit TRV this is a piece of s.... I do have this OTA issue as well. It just hangs in idle state and doesn't get any further.
Its not just that Eurotronic's QA has failed to an unbelievable degree. I mean these issues become visible almost instantly after pairing it with a coordinator. Further more it is the burden and duty of the end user to buy a ConBee/RaspBee, setup another Raspberry with deCONZ and hack around with the new but stale firmware. Unbelievable. Made in Germany in it's original sense.
I will send them back too. @dako76 any hints to be successful with that?

![image](https://user-images.githubusercontent.com/7912249/109348129-66baf500-7874-11eb-9db5-a740820c601b.png)

**Comment by dako76** [comment]
@pumuugel82 Hi, no they are very friendly, i send my Spirit to them and get a new one back. Just tell you have this issues, they know this :-) ...

**Comment by saxn-paule** [comment]
In my case they responded after more than one month and told the seller is responsible.

**Comment by iWelna** [comment]
@dako76  thx, i have already sent them an email with this problem, lets see what they will respond.   I want to buy 3 more but without working stuff i wont do it ... and didnt find any other solution with such quality tbh (maybe netatmo ? )

**Comment by univok** [comment]
Just a few questions because of my lack of understanding:

1. I can read the attribute "Date Code" in the Basic Cluster Info in the deCONZ UI, it is "20191014" for all my six Spirits. Where do I find the "Firmware" attribute like "22190930" mentioned in many of the posts here?

2. In the OTAU plugin of deCONZ, one of my six Spirits shows Vendor/Image/Version 0x1037/0x110c/0x0162e9d2. The other five show 0x0000 for all three values. If I press "Query", nothing changes. What does this mean?

3. Can I find at this location https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases offical release of the Spirit's firmware - or is it just a user compiled hack I better keep my fingers off?

**Comment by ebaauw** [comment]
1. _SW Build ID_ (0x4000);
2. That the _Query_ command doesn't reach the TRV or that the TRVs response doesn't reach the coordinator.
3. I don't know the formal status, but I installed this version on all my eight TRVs without any issue.  My TRVs had older firmware, dated 20181205; the faulty firmware doesn't want to be upgraded, see above.

**Comment by univok** [comment]
>     1. _SW Build ID_ (0x4000);
![grafik](https://user-images.githubusercontent.com/43472343/109390538-8c96d700-7912-11eb-8877-5b5ec3c57eb5.png)

Thank you for that hint! My six devices show nothing there, the value is empty. Good chance that I didn't pick up the faulty firmware with SW Build ID = 22190930 which can't be upgraded?

**Comment by ebaauw** [comment]
For (battery-powered) devices without _Receiver on When Idle_, the GUI only reads the attributes up to _Date Code_ when pressing _Read_.  For the other attributes, you need to double-click the attribute name to popup the _Attribute Editor_ window, and then press _Read_ in the popup window.  See the _User Manual_ (under _Help_).

**Comment by univok** [comment]
> you need to double-click the attribute name to popup the _Attribute Editor_ window, and then press _Read_ 

Thanks again. Now all devices show the fateful SW Build ID "22190930" weird! Interesting, because they don't seem to behave unique. For example, one of the six doesn't show "Off" in display when set to "Off" mode by using iOS Eve (possible with your awesome homebridge-hue!), the others do. It seems to me that Eurotronics wasn't very carefully maintaining build IDs.

I tried an update with 20190408_EUROTRONIC_Spirit_Zigbee_0x00122C380.ota to this Spirit, but it ended on IDLE after a few seconds. Vendor/Image/Version was  0xFFFF afterwards, but this was cleared to 0x0000 after restarting the GUI, a query has no effect as mentioned above.

**Comment by Witriol** [comment]
> 3. Can I find at this location https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/releases offical release of the Spirit's firmware - or is it just a user compiled hack I better keep my fingers off?

@univok 
3.  It is oficial firmware from Eurotronic I have received for publishing in here. It has been distributed with Spirit devices before the problematic fw with datecode 20191014.

**Comment by Witriol** [comment]
> @Witriol
> thanks for update regarding GitHub page maintaining.
> 
> How about firmware update for other device? See this [comment ](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/issues/1#issuecomment-768987989)above?
> Is there any other Firmware from Eurotronic? I also ordered two new devices, and one can't even be paired, other it does pair but have Firmware build date: 20191014 ; Firmware version: 22190930. zigbee2mqtt point this message Failed to call 'OTAUpdate' 'onZigbeeEvent' (AssertionError [ERR_ASSERTION]: No image available for imageType '65535'.

@vrabac Unfortunatelly, I do not know. I do not have any new update from Eurotronic.

**Comment by Chreece** [comment]
@Witriol 
Hello from me :)
Thank you for your time and patience answering all our questions. I know how it is to have to react with dissatisfied costumers, and also not having all the right answers for them...
I don't even know if you have an official role by Eurotronics.
As you mentioned, at beginning of March should be here an employee to maintain the repository (after many "next week or two more weeks").
Please don't misunderstand us, many of us didn't yet refund the faulty Thermostats because we believe in the product and think that it's only a software problem that could be easily updated and corrected at a given time.
We are like kids, if we been said "next week you will get what you want" we believe that and wait for it... If we don't get it we are upset! It's  also not a present for us but what we have paid for...

So to come back to the issues, I also have the "no update" devices, that also overheat or don't even reach the given temperature or even don't react by changing the offset. Also can't turn them off or use the HEAT mode.

And I still got them...

Sorry for my English, I hope you are well and forgive me referring to you as you are the company "Eurotronics" :)

**Comment by vespian** [comment]
Any update?

**Comment by iWelna** [comment]
I've got an email from Eurotronic to send them the broken termostat as i did. Now i'm waiting for the new one, lets hope it will work as it should.

**Comment by vrabac** [comment]
> 
> 
> I've got an email from Eurotronic to send them the broken termostat as i did. Now i'm waiting for the new one, lets hope it will work as it should.

I did the same, got offer from Amazon to return back or send them to Eurotronic, I did chose the second option as I need the TRVs.

**Comment by iWelna** [comment]
I've got today new termostat directly from Eurotronic, everything working as it should. Now i need 2 -5 more, wondering what soft will get ><

heres the info of what i had:

```
<OLD>
"model":"SPZB0001"
"dateCode":"20191014"
"softwareBuildID":"22190930"

<NEW>
"model":"SPZB0001"
"dateCode":"20190408"
"softwareBuildID":"18190930"
```

**Comment by ebaauw** [comment]
That's the same _Date Code_, but a slightly higher version that the firmware posted here (18181120).

**Comment by iWelna** [comment]
They have attached a letter with apologies and that this new spirit Zigbee is from "our current production batch"

Just made automations within hassio and can confirm TRV working perfectly fine.

**Comment by pumuugel82** [comment]
I received an Email from their support that I am allowed to send the faulty trvs back to them. I've not managed to do so but I am delighted now that the trvs will get fixed soon.

**Comment by oberschwabe** [comment]
same here : sent back faulty device with "dateCode":"20191014" after i contacted support via email and received already 4 days later a new device with "dateCode":"20190408".
Now everything works as expected - great support from Eurotronics

**Comment by double1968** [comment]
There is no possibility to update the firmware ??? Is it necessary to send the products to Eurotronics?
It seems crazy to me.

**Comment by Munvik** [comment]
Guys do u have 
Firmware build date : 20191014
Firwmare version : 22190930
Like me?
I can only publish occupied_heating_setpoint
Can't they make new commit to fix it? grrr

**Comment by Chreece** [comment]
> There is no possibility to update the firmware ??? Is it necessary to send the products to Eurotronics?
> It seems crazy to me.

As already mentioned, there are some firmwares that couldn't get updates. If you have one of the TRVs with datecode that matches one of them, you can only send them back. If you can already update yours with the firmware that you can find here, there is no reason to send them. 
This story with "new firmware comes soon" goes way back in December 2020.
We can only hope that eurotronic works on a stable new firmware, since there is no official statement about it...

**Comment by univok** [comment]
Today, one of my Spirits didn't open the valve at 15 °C room temperature though 20 °C was set. I could fix this by setting 0x4000 "TRV Mode" to "unknown 2" and 0x4001 "Set Valve Position" to 100 Percent. and then back to normal mode.

After this, I thought it is time now to write an email to EUROTRONIC TECHNOLOGY GMBH info@eurotronics.org, asking for replacement / repair at Eurotronics.

I'll keep you informed what they'll say.

**Comment by cleversmart** [comment]
just for info, my experience with the Spirits:
https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1098#issuecomment-800863193

**Comment by univok** [comment]
@saxn-paule 
> Also after removing/inserting batteries again it keeps flashing while the display shows nothing.

That's exactly how an OTA update (2h20min) from 20191014/22190930 ended for me today. The manual says that red flashing means an "Over the air update". It could be that the just finished data transfer is written to CPU EEPROM in this state? Just a guess. Fortunately it wasn't interrupted by taking out the batteries experimentally. After a few minutes, flashing stopped, showing "AdA" again. The unit was on 20190408/18181120 firmware then as intended.

**Comment by univok** [comment]
> I'll keep you informed what they'll say.

Within one hour, EUROtronic responded. They asked me to try the OTA update on all 6 Spirits, providing the well known links to Github. Fortunately, though showing the suspicous date code 20191014 and build 22190930, five of the six devices did the update to date code 20190408, this took the whole day (90 - 220 minutes per device), but one did not, showing IDLE, game over.

Worth mentioning that the deCONZ GUI OTAU plugin works a little bit fuzzy. One time it showed "Qeued" when pressing "Update", because there is only one update at a time, but mostly it didn't respond to the update button when there was another update in progress showing "IDLE", but starting anyway when the ongoing process had finished. It ended with either "done" or "no file", don't know why to distinguish between that. I think this is a little bit strange for a OTAU-plugin, because doing firmware updates is a delicate step.

I mailed back to EUROtronic and asked for the replacement of the one Spirit device that doesn't update. Keep you informed.

**Comment by vrabac** [comment]
> 
> 
> > I've got an email from Eurotronic to send them the broken termostat as i did. Now i'm waiting for the new one, lets hope it will work as it should.
> 
> I did the same, got offer from Amazon to return back or send them to Eurotronic, I did chose the second option as I need the TRVs.

Got my (2 peace) two days before from EUROtronic directly. They have following data:
```
Zigbee model: SPZB0001
Description: Spirit Zigbee wireless heater thermostat
Firmware build date: 20190408
Firmware version: 00000000
Vendor: Eurotronic
Model: SPZB0001
```
To mention only, that aftert I got message to Ins (install TRV on Radiator), which I did and press middle button, the device start blinking without any error message, i left it so for few minutes then it start doing something (motor) and got ER-1 error messaged, I pulled out battery and inserted them back, and when i get again Ins message i again press the middle button, and it passed it to Adr message, and after calibration it all went ok.

zigbee2mqtt schows now this for check ota:
```
Zigbee2MQTT:info Checking if update available for 'TermostatKinderzimmer'
Zigbee2MQTT:info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Checking if update available for 'TermostatKinderzimmer'","meta":{"device":"TermostatKinderzimmer","status":"checking_if_available"},"type":"ota_update"}'
Zigbee2MQTT:info No update available for 'TermostatKinderzimmer'
Zigbee2MQTT:info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"No update available for 'TermostatKinderzimmer'","meta":{"device":"TermostatKinderzimmer","status":"not_available"},"type":"ota_update"}'
Zigbee2MQTT:info MQTT publish: topic 'zigbee2mqtt/TermostatKinderzimmer', payload '{"current_heating_setpoint":5,"eurotronic_error_status":0,"eurotronic_host_flags":{"boost":false,"child_protection":true,"mirror_display":false,"window_open":false},"eurotronic_system_mode":129,"linkquality":73,"local_temperature":19.9,"occupied_heating_setpoint":21,"pi_heating_demand":0,"system_mode":"auto","unoccupied_heating_setpoint":16,"update":{"state":"idle"},"update_available":false}'
Zigbee2MQTT:info MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"TermostatKinderzimmer","updateAvailable":false},"status":"ok","transaction":"ez249-1"}'
```

**Comment by vrabac** [comment]
Does anyone know what is the: "eurotronic_system_mode":129
```
Zigbee2MQTT:info  2021-03-18 11:26:01: MQTT publish: topic 'zigbee2mqtt/TermostatKinderzimmer', payload '{"battery":80,"current_heating_setpoint":5,"eurotronic_error_status":0,"eurotronic_host_flags":{"boost":false,"child_protection":true,"mirror_display":false,"window_open":false},"eurotronic_system_mode":129,"linkquality":76,"local_temperature":18.22,"occupied_heating_setpoint":5,"pi_heating_demand":0,"system_mode":"auto","unoccupied_heating_setpoint":16,"update":{"state":"idle"},"update_available":false}'
Zigbee2MQTT:info  2021-03-18 11:26:07: MQTT publish: topic 'zigbee2mqtt/TermostatSchlafzimmer', payload '{"battery":100,"current_heating_setpoint":5,"eurotronic_error_status":0,"eurotronic_host_flags":{"boost":false,"child_protection":true,"mirror_display":false,"window_open":false},"eurotronic_system_mode":129,"linkquality":76,"local_temperature":19.54,"occupied_heating_setpoint":5,"pi_heating_demand":0,"system_mode":"auto","unoccupied_heating_setpoint":16,"update":{"state":"idle"},"update_available":false}'
```

I was thinking there are only three system mode:
modes: 
      - "off"
      - "auto"
      - "heat"

**Comment by elmarx** [comment]
@vrabac eurotronic system mode is a bitfield, which is superseded by "eurotronic_host_flags". See https://www.zigbee2mqtt.io/devices/SPZB0001.html#device-type-specific-configuration

**Comment by saxn-paule** [comment]
Tried to update one of my TRVs today after it was at 100% although the window was open and Home-Assistant said it is closed.
First try was stuck at 81% after 2 hours.
Second try completed but the version is stil the same.

What did I wrong? The TRV never started flashing red but the % display in Deconz counted from 0 to 100% within ~2hours

The third try doesn't start at all and is stil idle

![image](https://user-images.githubusercontent.com/8352690/111709490-f9cdc600-8847-11eb-8e9c-b0a85c14c8cd.png)

**Comment by bcutter** [comment]
> 
> 
> Tried to update one of my TRVs today after it was at 100% although the window was open and Home-Assistant said it is closed.
> First try was stuck at 81% after 2 hours.
> Second try completed but the version is stil the same.
> 
> What did I wrong? The TRV never started flashing red but the % display in Deconz counted from 0 to 100% within ~2hours
> 
> The third try doesn't start at all and is stil idle
> 
> ![image](https://user-images.githubusercontent.com/8352690/111709490-f9cdc600-8847-11eb-8e9c-b0a85c14c8cd.png)

Tried to update too, but update didn´t start at all, TRV/node stayed in "Idle" state.

**Comment by mm28ajos** [comment]
I was able to get my trvs with the faulty firmware replaced as well. OTA was also not possible for me. However, have to check the new one's.

**Comment by univok** [comment]
As reported before, I replaced one of my six Spirits by sending it to Eurotronic after contacting them by mail and checking implied warranty.  It was the one that didn't allow OTA updates. Eurotronic sent me a brandnew Spririt Zigbee in return without delay. I didn't look for the new firmware version yet, but supposedly it will work at least in terms of OTA updates now. 
EDIT: The new device has firmware Date Code=20190408, SW Build ID=18181120

**Comment by saxn-paule** [comment]
@Witriol are there any news from Eurotronic side? What about the new employee?

**Comment by Witriol** [comment]
> @Witriol are there any news from Eurotronic side? What about the new employee?

Unfortunately I have no news from Eurotronic still. My contact has been ill for some time now.

**Comment by pumuugel82** [comment]
I received 5 TRV today as a replacement for the faulty ones. I took about 1 week from giving the faulty to DHL and receive the new ones. Great customer care.!

**Comment by dpasdar** [comment]
I have also received two replacement units with the correct firmware, however after a couple of days of testing, the same problems started showing up again, apparently they lose connection to the hub and therefore let the valve open, you wake up with a sweat in the middle of the night struggling to find a way to turn it on/off again :(

**Comment by Chreece** [comment]
> I have also received two replacement units with the correct firmware, however after a couple of days of testing, the same problems started showing up again, apparently they lose connection to the hub and therefore let the valve open, you wake up with a sweat in the middle of the night struggling to find a way to turn it on/off again :(

Which firmware do they have?

**Comment by solvisit** [comment]
It is possible to replace the broken 20191014 firmware with a good 20191014 (or any other version), opening the TVR and programming the chip, it is not particularly difficult.
Steps:
1) You need a working TVR from which extract the working firmware or getting the file from someone did it. I will post the firmware if Eurotronic will permit it.
2) You have to download the flash programmer for JN5179: https://www.nxp.com/downloads/en/software-development-kits/JN-SW-4107.zip
3) You need an FTDI adapter ( maybe it works with a serial to usb cable too, didn't tested)
4) Open the TVR
5) Take off the zigbee board
6) Connect it to the FTDI (not connected to USB) (see the image for wiring, there is an old hw version with the connectors on the other side, they are just translated not mirrorerd)
7) Connect the FTDI to USB while shortcircuiting the board (see image)
8) Open a Windows console with "cmd" and go to the flash programmer directory (default:"C:\NXP\ProductionFlashProgrammer") 
8a) To extract the firmware launch the command (if i remember correctly, read the help for tips): JN51xxProgrammer.exe -s COM8 -S internal -F firmware.bin
8b) To program the firmware on the chip launch the command: JN51xxProgrammer.exe -s COM8 -S internal -f firmware.bin
You have to try a few times to get connection with the chip, sometimes it doesn't work at first try. The chip is entered in programming mode if it does not to send packets, you can check it on the FTDI leds.
![DSC_0041 1](https://user-images.githubusercontent.com/31163597/116807000-fdd05180-ab30-11eb-9caf-1e0176b87864.JPG)
![DSC_0042 1](https://user-images.githubusercontent.com/31163597/116807026-26f0e200-ab31-11eb-9cfa-a0d07b699af4.JPG)
![DSC_0044 1](https://user-images.githubusercontent.com/31163597/116807037-3d973900-ab31-11eb-8e37-09b20081c50a.JPG)
![DSC_0045 1](https://user-images.githubusercontent.com/31163597/116807080-79ca9980-ab31-11eb-9bf2-bcd7da8afce4.JPG)
![DSC_0047 1](https://user-images.githubusercontent.com/31163597/116807132-b39ba000-ab31-11eb-9ba0-b11ac21fb952.JPG)
![DSC_0052 1](https://user-images.githubusercontent.com/31163597/116807162-d75ee600-ab31-11eb-80e8-10e932507e3e.JPG)
![DSC_0053 1](https://user-images.githubusercontent.com/31163597/116807308-c2368700-ab32-11eb-9041-2a9592d37813.JPG)
![DSC_0055 1](https://user-images.githubusercontent.com/31163597/116807315-cf537600-ab32-11eb-9458-528961a5abe7.JPG)

**Comment by solvisit** [comment]
Forget the most important, how to short circuit the chip
![DSC_0048 1](https://user-images.githubusercontent.com/31163597/116807465-9bc51b80-ab33-11eb-842d-34a91dafe59d.JPG)

**Comment by solvisit** [comment]
I prefer to use the good 20191014  instead of the old 20190408, it seems more stable.

**Comment by ebaauw** [comment]
Do we have an OTA file for the good 20191014?

I'm running on the 20190408 version, which has connectivity issues on reboot, when connecting directly to the RaspBee / ConBee with firmware 0x26390500.

**Comment by solvisit** [comment]
No, we have only the OTA 20190408 , i have the good firmware 20191014 unecrypted for serial flash. We can ask to to Eurotronic to produce one OTA for it.

Yo are using a very old Conbee firmware, there is any particular reason?

**Comment by ebaauw** [comment]
That is the latest firmware, for the old (non-II) RaspBee and ConBee.

**Comment by Witriol** [comment]
Unfortunately I have only received OTA file with 20190408 version. I will try to ask Eurotronic for 20191014.

**Comment by solvisit** [comment]
I have the OTA :) I can build it from firmware

**Comment by solvisit** [comment]
I will ask Eurotronic if I can share, if not I will explain how to build it on your own

**Comment by solvisit** [comment]
I was wrong, cant build it. We have to ask Eurotronic for the OTA file.

**Comment by univok** [comment]
I wouldn't be surprised if the manufacturer discontinued the product soon instead of releasing better firmware.

**Comment by ghost** [comment]
An information for all fellow sufferers: After contacting Eurotronic support, they kindly replaced my devices with the faulty firmware version with new devices running on firmware 20190408.

**Comment by bcutter** [comment]
> 
> 
> An information for all fellow sufferers: After contacting Eurotronic support, they kindly replaced my devices with the faulty firmware version with new devices running on firmware 20190408.

Hopefully you won´t end up like https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/issues/1#issuecomment-825517339

Eurotronic should finally start fixing things instead of RMA-ing stuff continuously (which is a nice workaround, but not a solution).

**Comment by magicdude4eva** [comment]
I have now also contacted Eurotronic - 8 thermostats failed all at once today, disconnecting from the Zigbee network and opening the valves full blast. Frequently it is impossible to control the thermostats (using Homeassistant with Zigbee2mqtt) despite having 3 repeaters in the house. 

Feedback from Eurotronic:
> Wir freuen uns, Ihnen mitteilen zu können, dass das Over the Air Update für Spirit Zigbee nun verfügbar ist: https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/tree/main. Unter folgendem Link finden Sie eine Beschreibung zur erfolgreichen Installation: https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/wiki/OTA-update-quick-guide-via-deCONZ Sollte das Update bei Ihnen nicht funktionieren, bitte ich Sie uns Ihre Kaufbelege zukommen zu lassen.

Basically referring me to this Github project to flash the OTA update which I already did. I will now return all 8. Does anyone know if both Zigbee and Zwave are equally horrible? I would move to Zwave if the quality is better, otherwise a refund would be in order - blowing €320 on something which does not work is not sitting well.

**Comment by magicdude4eva** [comment]
I have returned all 8 TRVs and Eurotronic has sent my 8 replacements - packaging and TRV looks different to the previous model I purchased in October 2020:

<img width="598" alt="CleanShot 2021-06-18 at 15 44 35@2x" src="https://user-images.githubusercontent.com/1632781/122570506-1ac2d480-d04c-11eb-9709-5b9dd463fe87.png">

Right now we have scorching summer and I honestly can not test them at all.

**Comment by WolfRevo** [comment]
> 
> 
> Unfortunately I have only received OTA file with 20190408 version. I will try to ask Eurotronic for 20191014.

It has been a while or to say differently: it's now July. Has there been any feedback from Eurotronic on their intention to work on a new firmware or to release a debugged 20191014? It would be cool to have devices that won't loose connection over time before the next heating period starts.

**Comment by Witriol** [comment]
@WolfRevo 
Sorry, I do not have any news since I have received last email from Eurotronic in the beginning of May and since then they did not reply to any of our emails or telephone calls.

**Comment by WolfRevo** [comment]
That's a pitty as this feels like the company will eventually drop the support instead of fixing the issues... You are not responsible for this, I am aware.

**Comment by WolfRevo** [comment]
Short update: I ordered two Danfoss Ally TRVs. If you are using deCONZ or Zigbee2MQTT you can use them out of the box. They are well integrated. They don't loose any connections (so far I can tell), they accept external temperature setpoints (and use the mean of the internal and external temperature for temperature control) as well as open window detection.
Only if you use ZHA there is a slight issue I already filed a bug for in the [zigpy/zha-device-handlers](https://github.com/zigpy/zha-device-handlers/issues/1005) which causes the setpoint sent by HA to be processed by the internal PID mechanism so the valve opens and closes veeeeeeery slowly to adapt to the new setpoint (means the PID is not reset at the moment the setpoint is sent).
They are a little bit more expensive but they are nearly silent, the battery life is better and they work as intended. If the ZHA issue is closed fast enough and the two ones in test are working well for the next weeks I will replace my existing Eurotronic devices prior to the next heating period.

**Comment by bcutter** [comment]
I´m going totally crazy bout those devices. They worked pretty reliable when having them brand new in the first months of this year (well, almost - see https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1098#issuecomment-813060921). Now as heating period started I wondered why it´s either cold when it shouldn´t (heater off) or it´s too hot when it shouldn´t (heater on).

What I discovered:
1. Commands from Home Assistant sent via ZigBee controller (ConBee 2, integrated via deCONZ) are either never delivered to the TRV -or- the TRV doesn´t process them
2. Local changes (heating mode, temperature) are usually (not 100 %) sent back to the ZigBee controller and Home Assistant
3. The pre-configured reporting interval of 1200s is working "most of the time": temperature set in Home Assistant is updated by the local set one of the TRV after 20 minutes (sent to Home Assistant)

What I tried already:
- checked batteries: one is 45 %, the other one 70 %
- read all clusters including basic, power and thermostat
- Soft repairing (withouth deleting/removing them from ZigBee network) / factory reset according to manual with 3 button method
- hard repairing (complete deletion from deCONZ/removal from ZigBee network and rejoining with factory reset)
- (sent an eMail to info@eurotronic.org)

I have no idea what to try next. Is it
- TRV itself (I think so...)
- ZigBee mesh (it´s a strong one with several repeaters, so I doubt it)
- ZigBee master (ConBee 2)
- ZigBee software (deCONZ)
- ...?

Those TRVs are SO DAMN UNRELIABLE and there´re no other hardware alternative.
**--> Do you have any idea for me?
--> Is the behavior of my TRVs I reported typical / a well-known issue?**

**Comment by double1968** [comment]
@bcutter 

I had similar problems after the deconz 6.10.0 update.
Downgrading to deconz version 6.9.0 solved it for me. This appears to be a known problem.

**Comment by bcutter** [comment]
> @bcutter 
> 
> 
> 
> I had similar problems after the deconz 6.10.0 update.
> 
> Downgrading to deconz version 6.9.0 solved it for me. This appears to be a known problem.

Interesting. I use a docker container (addon in Home Assistant speak) and can't downgrade (or run beta releases).

So what did you do: did you file an issue in the deCONZ repo on GitHub? There is one, isn't it (you talked about a "known problem")? Or is your plan to just stay on that old version like forever?

**Comment by double1968** [comment]
I don't remember where I read it.
I haven't opened any issues on github.

**Comment by anbraten** [comment]
I had huge problems with binding my to Spirit Zigbee to my Conbee II stick. After changig some things it now works quite fine. What I changed:
- checking used wifi channels with an app an changig my zigbee stick to a rather free channel (used a mapping list of wifi vs zigbee channels)
- adding a lamp (router) to my network which seema to improve range drastically (even in my small flat)
- using an exteaion cord to move the conbee ii stick further away from my raspi with ssd

I can not say which change was important, but in the end I now can use my spirit zigbee   again and it seems to have a quite stable connection again.

**Comment by double1968** [comment]
I bought 4 Eurotronic Zigbee from the online shop.
I have no problems with pairing or loss of connection with my Spirits.
Only problem is:
Impossible to put the spirit OFF

My setup provides
RPI p3b +
Dongle Zwave
Dongle Conbee II - Deconz Plugin 6.9.0
SSD on USB port
HA core-2021.10.6

with the deCONZ 6.10.0 update
I lose the ability to control the temperature from HA but if I change the temperature on the valve it is transmitted to HA.

![Cattura3](https://user-images.githubusercontent.com/48420012/138691328-01a62f60-c704-4297-b929-18ebe8e38487.PNG)

Open Issue on Github
https://github.com/home-assistant/addons/issues/2244

**Comment by Se7enair** [comment]
> I bought 4 Eurotronic Zigbee from the online shop. I have no problems with pairing or loss of connection with my Spirits. Only problem is: Impossible to put the spirit OFF
> 
> My setup provides RPI p3b + Dongle Zwave Dongle Conbee II - Deconz Plugin 6.9.0 SSD on USB port HA core-2021.10.6
> 
> with the deCONZ 6.10.0 update I lose the ability to control the temperature from HA but if I change the temperature on the valve it is transmitted to HA.
> 
> ![Cattura3](https://user-images.githubusercontent.com/48420012/138691328-01a62f60-c704-4297-b929-18ebe8e38487.PNG)
> 
> Open Issue on Github [home-assistant/addons#2244](https://github.com/home-assistant/addons/issues/2244)

did you tried to turn the target-temperature to a value below the actual value?
I got two of these which are turning to `heat` and `off` automatically, depending if  the valve is open or not.
They are only support the states `heat` and `off` 

And i got a few others, which also support the state `auto`.
I can change them to `heat` and they will stay there. If i try to turn them `off` they change the temperature to 5°C and going back to `auto`.

The wierd thing, I got two thermostats which, obviously, have the same firmware, but one has `auto`-behaviour and one doesnt.

Thermostat 1 with `heat`, `off`
![grafik](https://user-images.githubusercontent.com/1680106/143848682-7439968b-aa44-4059-ad76-a69d62564da7.png)

Thermostat 2 with `heat`, `off`
![grafik](https://user-images.githubusercontent.com/1680106/143848754-ef7a254e-da09-4ef9-9624-5288649c390d.png)

Thermostat 3 with `auto`, `heat`, `off`
![grafik](https://user-images.githubusercontent.com/1680106/143848781-4e046499-6907-4301-bc28-4349e2830063.png)

As you can see, Thermostat 1 and 3, seems to be on the same firmware, but acting differently.

Can some explain this? And how can I bring all of them working the same?
Could this be related to the two versions of the 20191014 firmware?

**Comment by Se7enair** [comment]
Don't ask me why, but now all thermostat are working the same and owns the `auto`-mode.

**Comment by magicbenny** [comment]
> Don't ask me why, but now all thermostat are working the same and owns the `auto`-mode.

i have 3 of those TVRs. All of them have the same Firmware as yours and all of them Application Version 22. Whatever this means. However, all of them will be shown in HA with Auto Mode but as you described they will not accept temprature changes from HA :( Sometimes I thought that they will accept changes from Node-Red Automation but I obviously dreamed this. :(

**Comment by Se7enair** [comment]
> > Don't ask me why, but now all thermostat are working the same and owns the `auto`-mode.
> 
> i have 3 of those TVRs. All of them have the same Firmware as yours and all of them Application Version 22. Whatever this means. However, all of them will be shown in HA with Auto Mode but as you described they will not accept temprature changes from HA :( Sometimes I thought that they will accept changes from Node-Red Automation but I obviously dreamed this. :(

The problem with not accepting changes should be already solved. https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5433
Just wait for the next release or go back to v6.9

**Comment by dowhiletrue** [comment]
> It is possible to replace the broken 20191014 firmware with a good 20191014 (or any other version), opening the TVR and programming the chip, it is not particularly difficult. Steps:
> 
> 1. You need a working TVR from which extract the working firmware or getting the file from someone did it. I will post the firmware if Eurotronic will permit it.
> 2. You have to download the flash programmer for JN5179: https://www.nxp.com/downloads/en/software-development-kits/JN-SW-4107.zip
> 3. You need an FTDI adapter ( maybe it works with a serial to usb cable too, didn't tested)
> 4. Open the TVR
> 5. Take off the zigbee board
> 6. Connect it to the FTDI (not connected to USB) (see the image for wiring, there is an old hw version with the connectors on the other side, they are just translated not mirrorerd)
> 7. Connect the FTDI to USB while shortcircuiting the board (see image)
> 8. Open a Windows console with "cmd" and go to the flash programmer directory (default:"C:\NXP\ProductionFlashProgrammer")
>    8a) To extract the firmware launch the command (if i remember correctly, read the help for tips): JN51xxProgrammer.exe -s COM8 -S internal -F firmware.bin
>    8b) To program the firmware on the chip launch the command: JN51xxProgrammer.exe -s COM8 -S internal -f firmware.bin
>    You have to try a few times to get connection with the chip, sometimes it doesn't work at first try. The chip is entered in programming mode if it does not to send packets, you can check it on the FTDI leds.
>    ![DSC_0041 1](https://user-images.githubusercontent.com/31163597/116807000-fdd05180-ab30-11eb-9caf-1e0176b87864.JPG)
>    ![DSC_0042 1](https://user-images.githubusercontent.com/31163597/116807026-26f0e200-ab31-11eb-9cfa-a0d07b699af4.JPG)
>    ![DSC_0044 1](https://user-images.githubusercontent.com/31163597/116807037-3d973900-ab31-11eb-8e37-09b20081c50a.JPG)
>    ![DSC_0045 1](https://user-images.githubusercontent.com/31163597/116807080-79ca9980-ab31-11eb-9bf2-bcd7da8afce4.JPG)
>    ![DSC_0047 1](https://user-images.githubusercontent.com/31163597/116807132-b39ba000-ab31-11eb-9ba0-b11ac21fb952.JPG)
>    ![DSC_0052 1](https://user-images.githubusercontent.com/31163597/116807162-d75ee600-ab31-11eb-80e8-10e932507e3e.JPG)
>    ![DSC_0053 1](https://user-images.githubusercontent.com/31163597/116807308-c2368700-ab32-11eb-9041-2a9592d37813.JPG)
>    ![DSC_0055 1](https://user-images.githubusercontent.com/31163597/116807315-cf537600-ab32-11eb-9458-528961a5abe7.JPG)

I tried the solution @solvisit provided. I flashed the firmware, which runs on the working unit to the other. After that, the flashed unit just blinked red. Red blinking means OTA according to the manual. I then had to wait some minutes. Finally the broken-by-firmware unit worked like a charm.

Thanks @solvisit for your great work! Less electronic waste :-)

**Comment by bcutter** [comment]
Great if this worked for you.

At the end it's not a plug and play process and in general that's not what I expect of products still sold and still "supported": the need for hacking hardware like that on our own is just ridiculous. 

Would be so funny if not that sad.

**Comment by Se7enair** [comment]
> Great if this worked for you.
> 
> At the end it's not a plug and play process and in general that's not what I expect of products still sold and still "supported": the need for hacking hardware like that on our own is just ridiculous. 
> 
> Would be so funny if not that sad.

I have to contradict. Just recieved a few brand new thermostats. They work as expected and seem to have revised firmware. Datecode 20191014 Version 22190930.

**Comment by maody** [comment]
I have 6 Eurotronic SPZB0001 with 3 different FWs:  Version 22190930 Datecode 20191014, Version 18181120 Datecode 20190408 and Version 15181120 Datecode 20181205.

For me the oldest FW is the most reliable. My 2 Devices with Version 22190930 Datecode 20191014 are almost useless. They lose contact to broker, drain battery in very short time and open/close the valve without reason. 

I would like to flash the FW by myself, but how/where do i get a working 20191014 FW File?

**Comment by Se7enair** [comment]
Take a Look at the end of this issue . I posted the Firmware and added a few points to a already existing Tutorial.

https://github.com/KartoffelToby/ai_thermostat/issues/74#issuecomment-992754294

This is downloaded from a recently bought thermostat and should  bei the newest.

**Comment by Valbx** [comment]
Hello, I'm trying to flash my TVR with but I have no communication when I plug the USB. There is light, I can see the COM on my computer, but nothing else. I think I'm missing something. I'm using a CP210x to UART adaptor.

Question to those who succeeded in the operation : could you give more information about the point "7.Connect the FTDI to USB while shortcircuiting the board (see image)" please ? What do we have to short-circuit on the board ?

Already thanks for the hope :-)

**Comment by Se7enair** [comment]
There should be two rectangled contacts on the back side. Connect them while you connect the FTDI-Adapter to your computer.
Use anything that can conduct electricity. Even your finger should work.

https://user-images.githubusercontent.com/31163597/116807465-9bc51b80-ab33-11eb-842d-34a91dafe59d.JPG

**Comment by Valbx** [comment]
Hello. Working now :-) Thanks to the tips from Se7enair about reset with the 2 contacts on the back. But not on the first try.
I also had some problems with the baud rate in JN51xxProgrammer. So I used the command "JN51xxProgrammer.exe -s COM3 -P 115200 -S internal -f firmware.bin" and everything worked fine.

**Comment by sygys** [comment]
It seems that my board has its pins on the other side... not really sure if everything should be mirrored or how to proceed. also the pin layout is smaller then the standard jumper cables in which they bend. I cant really tell if i connect it wrong or that they dont make contact. i could solder some wires to the other side. but im not sure if that will work.

**Comment by Se7enair** [comment]
The connectors are translated, not mirrored. So same order of the cables. 

The pin layout is a little bit smaller. But I managed to get it connected to standard jumper cables. Gently bend the outer pins a little bit.

**Comment by sygys** [comment]
By the way you can also solder 4 wires to the other side. they both 6 pins seem to do the same thing. i flashed the firmware. i hope the crashes are gone now.. but im afraid all firmware versions are equally bad

**Comment by brozikcz** [comment]
Could someone confirm that some firmware works with manual TRV mode (mode:1) and set valve position?

**Comment by rusitschka** [comment]
Yes, firmware build date 20190408, firmware version 18181120, works but mode resets a couple of times per week and you have to set it again.

**Comment by brozikcz** [comment]
Unfortunately not for me.
I updated my device to 20190408 - 18181120 firmware, reset it, newly joined it to my zigbee network, and set trv mode into manual with 50% valve position but after few minutes (~2min) the device switched back to auto mode and after that, it's not possible to switch back to manual mode.

**Comment by rusitschka** [comment]
I set target temperature to 30 degrees before setting mode to manual. Maybe that's why it's working better for me?

**Comment by brozikcz** [comment]
Hm, probably no, because it again switches to auto mode.

**Comment by sygys** [comment]
In the end I tried everything. I tested all firmwares, different routers, different setting but this is not the problem. The devices loose their connection and the connection is then reset. This could be to a router change or maybe the firmware just crashes. 

In the end I bought 12 of the zwave variants. They have been running for a few months now in valve mode without any issues. I don't even need additional nodes. Zwave has an idiotic high range. I have reception with just 1 zwave stick In the middle of the house. This is a rock solid solution. Not once had problems with the eurotronic zwave valves.

**Comment by demey** [comment]
A downloaded firmware from one of my TRV's.
Firmware build date: 20191014
Firmware version: 22190930
Maybe it help to somebody.
[firmware_20191014_22190930.zip](https://github.com/EUROTRONIC-Technology/Spirit-ZigBee/files/9365754/firmware_20191014_22190930.zip)

**Comment by sygys** [comment]
Buy the zwave equivalent. The zwave valves fr euro tronic have been rock solid for 1 year straight. Not once have they lost connection or rebooted. Another great feature is that zwave has an extreme high range I have a zwave stick central in our home and without any additional routers I have them all connected to this stick.

**Comment by sygys** [comment]
Batteries of these things are insane too. 1 year in use with direct valve control (have to configure that manually) and they are at 90 - 95% battery

**Comment by bcutter** [comment]
Summary of two former posts:
"The product(s firmware) is crap. Buy another one from the same vendor. Get also another coordinator."

Nice 😊

**Comment by WolfRevo** [comment]
> Short update: I ordered two Danfoss Ally TRVs. If you are using deCONZ or Zigbee2MQTT you can use them out of the box. They are well integrated. They don't loose any connections (so far I can tell), they accept external temperature setpoints (and use the mean of the internal and external temperature for temperature control) as well as open window detection.
> Only if you use ZHA there is a slight issue I already filed a bug for in the [zigpy/zha-device-handlers](https://github.com/zigpy/zha-device-handlers/issues/1005) which causes the setpoint sent by HA to be processed by the internal PID mechanism so the valve opens and closes veeeeeeery slowly to adapt to the new setpoint (means the PID is not reset at the moment the setpoint is sent).
> They are a little bit more expensive but they are nearly silent, the battery life is better and they work as intended. If the ZHA issue is closed fast enough and the two ones in test are working well for the next weeks I will replace my existing Eurotronic devices prior to the next heating period.

Just to mention it again... I am totally happy with the Danfoss Ally Zigbee thermostats for over a year now. PID topic has been solved quiet fast. They do all the stuff the Eurotronics promised and they are rock solid. Distance isn't an issue with the Tradfri extender from IKEA also in very large houses.

**Comment by sygys** [comment]
If you use them in direct valve mode they will crash almost once a day. If you don't monitor them and use them in normal mode you will never see them crashing in direct valve mode you see this because the setpoint temperature will reset back for a second and it will revert back to normal mode. I tried at least 4 different firmware versions and all have the same issue.

These valves with Zigbee are the most irreliables I have ever seen on the other hand the zwave variants are rock solid like I said

**Comment by brozikcz** [comment]
Does anybody know if any other zigbee device exists with configurable `valve_position`?
I found just https://www.zigbee2mqtt.io/devices/106-03.html but that isn't available in my country.

**Comment by sygys** [comment]
I haven't found any. I bought the zwave variant from eurotronic. These are amazing. They have an insane range so no routers needed. I have 12 valves and only 1 USB stich to connect them. I recommend the zwave variants

**Comment by rusitschka** [comment]
> Does anybody know if any other zigbee device exists with configurable `valve_position`? I found just https://www.zigbee2mqtt.io/devices/106-03.html but that isn't available in my country.

Not Zigbee but very reliable using a trick: Homematic. Check out https://github.com/rusitschka/valves
